engine-adapter/camunda-platform-7-embedded-core/src/main/kotlin/dev/bpmcrafters/processengineapi/adapter/c7/embedded/task/delivery/pull/CachingProcessDefinitionKeyResolver.kt

```diff
@@ -0,0 +1,28 @@
+package dev.bpmcrafters.processengineapi.adapter.c7.embedded.task.delivery.pull
+
+import org.camunda.bpm.engine.RepositoryService
+
+data class CachingProcessDefinitionKeyResolver(
+  val repositoryService: RepositoryService,
+  private val keys: MutableMap<String, String> = mutableMapOf()
```

Comment:
With inline value classes, it would be easier to see what you save there exactly.